api/utils/keys/policy.go

```diff
@@ -72,10 +72,14 @@ func NewPrivateKeyPolicyError(p PrivateKeyPolicy) error {
 // ParsePrivateKeyPolicyError checks if the given error is a private key policy
 // error and returns the contained unmet PrivateKeyPolicy.
 func ParsePrivateKeyPolicyError(err error) (PrivateKeyPolicy, error) {
+	if err == nil {
+		return "", trace.BadParameter("provided error is not a key policy error: %v", err)
+	}
+
 	// subMatches should have two groups - the full string and the policy "(\w+)"
 	subMatches := privateKeyPolicyErrRegex.FindStringSubmatch(err.Error())
 	if subMatches == nil || len(subMatches) != 2 {
-		return "", trace.BadParameter("provided error is not a key policy error")
+		return "", trace.BadParameter("provided error is not a key policy error: %v", err)
```

Comment:
Would it make sense to use a slightly different error message in both of these cases? It might make troubleshooting this error easier.